### DISCUSSION TITLE
Use a secondary tile source for a specific bbox only

### DISCUSSION THREAD

**Discussion by laem** [maplibre/maplibre-gl-js#4491] [Q&A] [answered]
Hi, I'm wondering if it's possible to load a planet source as a base vector tile source, and then override it just for a local bbox. 

For instance, using a planet source provided by maptiler, but serve the specific tiles of France with another source that would be tailored for the particularities of the country. 

The "bounds" option lets us load the specialized tile source only for a specific bbox, but I couldn't find an inverted option to *not load* the planet source inside of the same bbox. 

Is it possible ? Would it be simple to extend MapLibre with this feature ? Could it be done with a protocol ? 

Regards

**Comment by HarelM** [comment]
I think you should be able to do it with addProtocol by checking the x y z and deciding which end point to fetch from.

**Reply by laem** [reply]
Thanks ! I looked for a list of protocol extensions but couldn't find a similar one. I'll try to build it myself.

**Comment by laem** [comment, marked as answer]
Done in https://github.com/cartesapp/cartes/blob/master/components/map/CartesProtocol.ts
